### PR TITLE
merge events

### LINKED ISSUE #68: merge CheckpointFinalized and LogCheckpoint
merge CheckpointFinalized and LogCheckpoint. add StateUpdate to Checkpoint.

### PR DESCRIPTION
**Pull request by tkmct**
merge `CheckpointFinalized` and `LogCheckpoint`.

close #68

### CODE REVIEW

File: contracts/DepositContract.sol [hunk 1]
```diff
@@ -109,16 +106,28 @@ contract DepositContract {
...
     function finalizeCheckpoint(types.Property memory _checkpointProperty) public {
         require(universalAdjudicationContract.isDecided(_checkpointProperty), "Checkpointing claim must be decided");
-        types.Checkpoint memory checkpoint = deserializeCheckpoint(_checkpointProperty);
+        // types.Checkpoint memory checkpoint = createCheckpoint(_checkpointProperty);
+        types.Range memory range = abi.decode(_checkpointProperty.inputs[0], (types.Range));
+        types.Property memory property = abi.decode(_checkpointProperty.inputs[1], (types.Property));
+        types.Checkpoint memory checkpoint = types.Checkpoint({
+            subrange: range,
+            stateUpdate: types.StateUpdate({
+                stateObject: property,
```
Comment (syuhei176) [inline contracts/DepositContract.sol]: This `property` is StateUpdate.

File: contracts/DepositContract.sol [hunk 2]
```diff
@@ -109,16 +106,28 @@ contract DepositContract {
...
+                range: range,
+                blockNumber: getLatestPlasmaBlockNumber() - 1,
```
Comment (syuhei176) [inline contracts/DepositContract.sol]: this should be inputs[2] of `checkpointProperty.inputs[1]`

File: contracts/DepositContract.sol (hunk 1 above)
Comment (tkmct) [reply contracts/DepositContract.sol]: so I just can use `_checkpointProperty.inputs[1]` as stateUpdate here?

Comment (syuhei176) [reply contracts/DepositContract.sol]: maybe it's reasonable.

**Pull request merged**